### PR TITLE
[iOS] Platform specifics to allow visual elements to become first responder

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

#20 has added `CancelsTouchesInView = false` in order to allow custom renderers to process gestures; however, this has the adverse affect of subviews not canceling tap gestures at page level. In turn, `View.EndEditing(true)` is called and results in regression as explained in the below bug description.

Added a platform specific option for any visual element to become first responder so that `ShouldReceiveTouch` of the page gesture recognizer can be set to false. This is useful in cases such as when a chat application does not need to dismiss keyboard when a `Send` button is hit.

### Bugs Fixed ###

- https://bugzilla.xamarin.com/show_bug.cgi?id=45723

### API Changes ###

Added:
 - `CanBecomeFirstResponder`
 - Getters/Setters for `CanBecomeFirstResponder`

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs [hunk 1]
```diff
@@ -95,12 +95,12 @@ public override void ViewDidLoad()
...
-			uiTapGestureRecognizer.ShouldRecognizeSimultaneously = (recognizer, gestureRecognizer) => true;
-			uiTapGestureRecognizer.ShouldReceiveTouch = OnShouldReceiveTouch;
-			uiTapGestureRecognizer.DelaysTouchesBegan =
-				uiTapGestureRecognizer.DelaysTouchesEnded = uiTapGestureRecognizer.CancelsTouchesInView = false;
+			var uiTapGestureRecognizer = new UITapGestureRecognizer(a => View.EndEditing(true))
+			{
+				ShouldRecognizeSimultaneously = (recognizer, gestureRecognizer) => true,
+				ShouldReceiveTouch = OnShouldReceiveTouch,
+				CancelsTouchesInView = false
+			};
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs]: Default for `DelaysTouchesBegan` and `DelaysTouchesEnded` is false already.

**Comment by rmarinho** [comment]
needs rebase

**Comment by adrianknight89** [comment]
Done.

**Comment by rmarinho** [comment]
needs docs update.

**Comment by adrianknight89** [comment]
Doesn't it have already? See `VisualElement.xml`.

**Comment by rmarinho** [comment]
Can you rebase to try again please @adrianknight89  ?

**Comment by adrianknight89** [comment]
Done.

**Comment by rmarinho** [comment]
Needs rebase

**Comment by adrianknight89** [comment]
Done.

**Comment by rmarinho** [comment]
You need to check the MacOS platform, sorry.. 

`C:\BuildAgent3\work\b1cbc78a4a5549fc\Xamarin.Forms.Platform.iOS\VisualElementRenderer.cs(112, 24): error CS0115: 'VisualElementRenderer<TElement>.CanBecomeFirstResponder': no suitable method found to override `

**Comment by adrianknight89** [comment]
@rmarinho Added `MOBILE` tag to make it available for iOS only. The property is not available in MacOS, and I'm not sure if this issue exists there.

**Comment by rmarinho** [comment]
build

**Comment by rmarinho** [comment]
Hey @adrianknight89 we are trying to clean some of these old pr's you mind rebase this and update the docs and reopen when that's done?!

Thanks

**Pull request closed without merging**